models/issue_content_history.go

```
@@ -0,0 +1,177 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"fmt"
+
+	"code.gitea.io/gitea/models/db"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/timeutil"
+
+	"xorm.io/builder"
+)
+
+// IssueContentHistory save issue/comment content history revisions.
+type IssueContentHistory struct {
+	ID             int64              `xorm:"pk autoincr"`
+	PosterID       int64              `xorm:""`
+	IssueID        int64              `xorm:"INDEX"`
+	CommentID      int64              `xorm:"INDEX"`
+	EditedUnix     timeutil.TimeStamp `xorm:"INDEX"`
+	ContentText    string             `xorm:"LONGTEXT"`
+	IsFirstCreated bool               `xorm:""`
+	IsDeleted      bool               `xorm:""`
+}
+
+func init() {
+	db.RegisterModel(new(IssueContentHistory))
+}
+
+// SaveIssueContentHistory save history
+func SaveIssueContentHistory(e db.Engine, posterID, issueID, commentID int64, editTime timeutil.TimeStamp, contentText string, isFirstCreated bool) {
+	ch := &IssueContentHistory{
+		PosterID:       posterID,
+		IssueID:        issueID,
+		CommentID:      commentID,
+		ContentText:    contentText,
+		EditedUnix:     editTime,
+		IsFirstCreated: isFirstCreated,
+	}
+	_, err := e.Insert(ch)
+	if err != nil {
+		log.Error("can not save issue content history. err=%v", err)
```

Could we ignore the error outside?